src/cpp/shared/game/game.cpp

```diff
@@ -165,4 +182,77 @@ void philAI::RedistributeTroops(armyGroup *army1, armyGroup *army2, int a1, int
 	}
+}
+
+void game::SetRandomHeroArmies(int heroIdx, int isAI) {
+  int randomizedValue; // eax@1
+  int randomizedValue2;
+  int creatureTier2 = 0; // [sp+10h] [bp-8Ch]@1
+  int randomUpperBound; // [sp+14h] [bp-88h]@9
+  signed int armySlotIdx; // [sp+18h] [bp-84h]@3
+  signed int creatureTier; // [sp+18h] [bp-84h]@6
+  int randomLowerBound; // [sp+1Ch] [bp-80h]@9
+                        //__int16 creatureTypesAndQuantities[54]; // [sp+20h] [bp-7Ch]@1
+  armyGroup *pointerToArmyOfHeroIdx = &gpGame->heroes[heroIdx].army; // [sp+8Ch] [bp-10h]@1
+  bool hasTier[3]; // [sp+90h] [bp-Ch]@1
+  int TIER_ONE = 0;
+  int TIER_TWO = 1;
+  int TIER_THREE = 2;
+
+  hasTier[TIER_ONE] = 1;
+  hasTier[TIER_TWO] = Random(0, 99) < (isAI < 1 ? 50 : 80);// If isAI, there is a ~20% chance that this hero will get Tier 2 creatures; if not, then a ~50% chance
+  randomizedValue2 = Random(0, 99);
+  hasTier[TIER_THREE] = (isAI < 1 ? 25 : 65) > randomizedValue2;// If isAI, there is a ~65% chance that this hero will get Tier 3 creatures; if not, then a ~25% chance
+  if ((isAI < 1 ? 25 : 65) <= randomizedValue2)// If isAI, there is a ~35% chance that this hero will get Tier 2 creatures; if not, then a ~75% chance
+    hasTier[TIER_TWO] = 1;
+  for (armySlotIdx = 0; armySlotIdx < 5; ++armySlotIdx) {
+    pointerToArmyOfHeroIdx->creatureTypes[armySlotIdx] = -1;
+    pointerToArmyOfHeroIdx->quantities[armySlotIdx] = -1;
+  }
+  for (creatureTier = 0; creatureTier < 2; ++creatureTier) {
+    if (hasTier[creatureTier]) {
+      pointerToArmyOfHeroIdx->creatureTypes[creatureTier2] = creatureTypesAndQuantities[gpGame->heroes[heroIdx].factionID][creatureTier][TIER_ONE];
+      randomLowerBound = 10 * creatureTypesAndQuantities[gpGame->heroes[heroIdx].factionID][creatureTier][TIER_TWO];
+      randomUpperBound = 10 * creatureTypesAndQuantities[gpGame->heroes[heroIdx].factionID][creatureTier][TIER_THREE] + 9;
+      if (isAI) {                              //  If isAI, randomLowerBound is assigned the average of the bounds and results in the probability of higher random values
+        randomLowerBound = (randomUpperBound + randomLowerBound) / 2;
+      }
+      randomizedValue = Random(randomLowerBound, randomUpperBound) / 10;
+      pointerToArmyOfHeroIdx->quantities[creatureTier2++] = randomizedValue;
+    }
+  }
+}
+
+void game::GiveTroopsToNeutralTown(int castleIdx) {
+  int randomQuantity; // [sp+18h] [bp-14h]@7
+  int quantity; // [sp+18h] [bp-14h]@15
+  signed int creatureTier; // [sp+1Ch] [bp-10h]@7
+  int creatureType; // [sp+24h] [bp-8h]@0
+  int creatureTierRandomizer; // [sp+28h] [bp-4h]@4
+
+  if ((this->castles[castleIdx].x > 0 || this->castles[castleIdx].y > 0) && this->castles[castleIdx].ownerIdx < 0) {
+    creatureTierRandomizer = Random(1, 15);
+    if (giCurTurn / 10) { // As time goes on, the odds of getting smaller stacks of stronger creatures increases, but it is still possible to get larger stacks of weaker creatures
```

Comment:
This should be "giCurTurn / 10 > 0".

Reply:
...interestingly enough, this is what it decompiled as, which may be concerning. Also, I'm not sure what changing it to "giCurTurn / 10 > 0" would accomplish. I don't quite understand it.

Reply:
Yeah, compilers optimize "if (x != 0)" to "if (x)".  They're equivalent.

giCurTurn / 10 > 0 is the same thing as what's currently there,  but more clear.

Reply:
Ah, ok.